src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs

```diff
@@ -54,30 +58,30 @@ public class CookieAuthenticationEvents : ICookieAuthenticationEvents
         /// Implements the interface method by invoking the related delegate method
         /// </summary>
         /// <param name="context"></param>
-        public virtual void ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
+        public virtual Task ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
```

Comment:
You changed the return value here, but you didn't add any await's at the call sites in the Handler.